lib/dal/src/component.rs

```diff
@@ -1403,6 +1579,59 @@ impl Component {
 
         Ok(results)
     }
+
+    pub async fn copy_paste(&self, ctx: &DalContext, offset: (f64, f64)) -> ComponentResult<Self> {
+        let schema_variant = self.schema_variant(ctx).await?;
+
+        let mut pasted_comp = Component::new(ctx, "Copy".to_owned(), schema_variant.id()).await?;
+
+        let x: f64 = self.x().parse()?;
+        let y: f64 = self.y().parse()?;
+        dbg!(x, y, offset, self.width(), self.height());
+        pasted_comp
+            .set_geometry(
+                ctx,
+                (x + offset.0).to_string(),
+                (y + offset.1).to_string(),
+                self.width(),
+                self.height(),
+            )
+            .await?;
+
+        pasted_comp.clone_attributes_from(ctx, self.id()).await?;
+
+        pasted_comp
+            .set_resource(
+                ctx,
+                DeprecatedActionRunResult {
+                    status: Some(ResourceStatus::Ok),
+                    payload: None,
+                    message: None,
+                    logs: Vec::new(),
+                    last_synced: Some(Utc::now().to_rfc3339()),
+                },
+            )
+            .await?;
+
+        pasted_comp
+            .set_name(ctx, &format!("{} - Copy", self.name(ctx).await?))
+            .await?;
```

Comment:
I could be wrong, but won't the component's name be `Copy - Copy` with how this is written?